deepchem/data/dft_data.py

```diff
@@ -0,0 +1,375 @@
+
+    @property
+    def entry_type(self) -> str:
+        return "dens"
+
+    def get_true_val(self) -> np.ndarray:
+        dens = np.load(self.entry_dct["trueval"])
+        return dens
+
+    def get_val(self, qcs: List[KSCalc]) -> np.ndarray:
+        """
+        This method calculates the integration grid which is then used to calculate the
+        density profile of an entry object.
+
+        Parameters
+        ----------
+        qcs: List[KSCalc]
+
+        Returns
+        -------
+        Numpy array containing calculated density profile
+        """
+        qc = qcs[0]
+
+        grid = self._get_integration_grid()
+        rgrid = grid.get_rgrid()
+
+        return (qc.dens(rgrid)).numpy()
+
+    def get_integration_grid(self) -> BaseGrid:
+        """
+        This method is used to calculate the integration grid required for a system
+        in order to calculate it's density profile using Differentiable DFT.
+
+        Returns
+        -------
+        grid: BaseGrid
+
+        References
+        ----------
+        https://github.com/diffqc/dqc/blob/0fe821fc92cb3457fb14f6dff0c223641c514ddb/dqc/grid/base_grid.py
+        """
+        if self._grid is None:
+            system = self.get_systems()[0]
+
+            # reduce the level of grid
+            orig_grid_level: Optional[int] = None
+            if "grid" in system:
+                orig_grid_level = system["grid"]
+
+            # get the dqc grid
+            system["grid"] = "sg2"
+            dqc_mol = system.get_dqc_system()
+            dqc_mol.setup_grid()
+            grid = dqc_mol.get_grid()
+            assert grid.coord_type == "cart"
+
+            # restore the grid level
+            if orig_grid_level is not None:
+                system["grid"] = orig_grid_level
+            else:
+                system.pop("grid")
+
+            self._grid = grid
+
+        return self._grid
+
+
+class _EntryIE(DFTEntry):
+    """
+    Entry for Ionization Energy (IE)
+    """
+
+    def __init__(self, entry_dct):
+        """
+        Parameters
+        ----------
+        entry_dct: Dict
+
+        """
+        super().__init__(entry_dct)
+        self.entry_dct = entry_dct
+
+    @property
+    def entry_type(self) -> str:
+        return "ie"
+
+    def get_true_val(self) -> np.ndarray:
+        return (self.entry_dct["true_val"])
+
+    def get_val(self, qcs: List[KSCalc]) -> np.ndarray:
+        """
+        This method calculates the energy of an entry based on the systems and command present
+        in the data object. For example; for a Lithium hydride molecule the total energy
+        would be ; energy(Li) + energy(H) - energy(LiH)
+
+        Parameters
+        ----------
+        qcs: List[KSCalc]
+
+        Returns
+        -------
+        Total Energy of a data object for entry types IE and AE
+        """
+        glob = {"systems": qcs, "energy": self.energy}
+        return eval(self["cmd"], glob)
```

Comment:
Why does this work? This is no longer a dictionary so self["cmd"] shouldn't work. I think this might be broken

Reply:
In general calling eval() is dangerous and shouldn't be done I think